docs/en/document/developer-guide/xds-discovery.md

```diff
@@ -0,0 +1,119 @@
+
+## Feature Introduction
+
+**Service discovery based on xDS protocol** allows Sermant to connect to Istio's control plane to obtain Kubernetes service information and the corresponding specific service instance information.
+
+> Note: Using Sermant's xDS service discovery capability requires the service to be deployed in a Kubernetes container environment. 
+
+## Development Example
+
+This development example is based on the project created in the [Creating Your First Plugin](README.md) document and demonstrates how the plugin uses the xDS service discovery capability provided by the Sermant framework to obtain service instances:
+
+1. Add the `xdsServiceDiscovery` variable in the `com.huaweicloud.sermant.template.TemplateDeclarer` class under `template\template-plugin` in the project to get the xDS service discovery service provided by the Sermant framework for obtaining service instances:
+
+   ```java
+   XdsServiceDiscovery xdsServiceDiscovery = ServiceManager.getService(XdsCoreService.class).getXdsServiceDiscovery();
+   ```
+
+2. After obtaining the service discovery instance, you can call the APIs provided by `XdsServiceDiscovery` to perform corresponding actions. This example demonstrates how to directly obtain service instance information for a service named `service-test`:
+
+   ```java
+   @Override
+   public ExecuteContext before(ExecuteContext context) throws Exception {
+     Set<ServiceInstance> serviceInstance = xdsServiceDiscovery.getServiceInstance("service-test");
+     Iterator<ServiceInstance> iterator = serviceInstance.iterator();
+     while (iterator.hasNext()) {
+         ServiceInstance instance = iterator.next();
+         System.out.println("ServiceInstance: [" + instance.getHost() + " : "
+                 + instance.getPort() + "]");
+     }
+   }
+   ```
+
+After completing the development, refer to the [Packaging and Building](README.md#Packaging and Building) process in the Creating Your First Plugin document. Execute **mvn package** in the root directory of the project to generate the build artifact.
+
+   > Note: `service-test` must be the name of the Kubernetes service. The obtained service instance information corresponds to the pod instance data of that service.
+
+4. Enable the xDS service and set it to be enabled in `agent/config/config.properties` with the following configuration:
+   ```properties
+   # xDS service switch
+   agent.service.xds.service.enable=true
+   ```
+
+5. After completing the execution, package the Sermant image and the host microservice image. Start the `service-test`service in the k8s environment and create the corresponding service instances (implemented by the user). Finally, start the host application in Kubernetes and mount Sermant. Refer to the [Sermant Injector User Guide](../user-guide/sermant-injector.md#Startup and Result Verification) for guidance on packaging Sermant and the host image in the Kubernetes environment, as well as mounting Sermant for startup.
```

Comment:
Format Error ： [Sermant Injector User Guide](../user-guide/sermant-injector.md#Startup and Result Verification)